book/integration-service.md

```diff
@@ -0,0 +1,271 @@
+
+
+### Goals
+
+- We need to be able to deploy a specific set of images to an environment and record that result somewhere.
+- Given a specific set of images, you can tell if they passed application validation tests or not
+- We must have some way of tracking a specific set of images where we can update one component image at a time in order to support 'slow loop' application validation testing.
+- To be SLSA compliant, we should also be able to tell what version of the build and test pipelines were used to create each component image.
+
+The diagram below shows the interaction of the integration service and other services.
+
+![](../diagrams/hacbs-data-flow.jpg)
+
+
+### Dependencies
+
+The [Integration Service](./integration-service.md) is dependent on the following services:
+- [Pipeline Service](./pipeline-service.md)
+  - Pipeline execution, Pipeline logging
+- [GitOps Service](./gitops-service.md)
+  - Provides the facility to create
+    - Snapshots used to test Builds
+    - Environment to test the Application on
+    - SnapshotEnvironmentBindings to have the Snapshot deployed to a specific Environment
+- [Hybrid Application Service](./hybrid-application-service.md)
+  - Provides the Application and Component model. Integration Service updates the pullspec reference on the Component CR when a component passes its required testing.
+- [Release Service](./release-service.md)
+  - Provides the ReleasePlan that will be used to determine if integration-service should create a Release
+- [Enterprise Contract Service](./enterprise-contract.md)
+  - Provides facilities to validate whether content has passed the Enterprise Contract.
+
+
+## Terminology
+
+**Prelim Pipeline** - This is the test pipeline run for the Component of an Application that gets triggered by a completed Build pipeline. Deprecated name, refer to the Component Pipeline
+
+**Component Pipeline** - This is the same as the Prelim pipeline. Going forward the Prelim Pipeline will be referred to as Component Pipeline
+
+**Final Pipeline** - This is the pipeline run that runs if two or more Components have completed and passed their Component pipeline concurrently so that they should be tested together. Deprecated name, refer to Composite Pipeline.
+
+**Composite Pipeline** - This is the same as the Final Pipeline. Going forward the Final Pipeline will be referred to as Composite Pipeline
+
+**Global Candidate List** - The list of all Component digest that have passed individual component testing. This can be retrieved from the Application to see what Components it is made of and then querying each of the Components `spec.containerImage`. 
+
+**Snapshot** - The list that maps all Components of an Application to their Component Image digests. Once created, the snapshot is immutable. 
+
+**Snasphot Digest** - The snapshot itself should have an identifier constructed from its contents - using a digest of the contents likely makes sense.
```

Comment:
Can you tell us what the digest is instead of what should happen or what makes sense?

Reply:
This is actually a proposed item that we've been designing to optimize finding existing snapshots on the cluster.
It's not implemented yet, so I think we can drop this line entirely until we implement it.